src/ape/api/providers.py

```diff
@@ -681,16 +681,16 @@ def estimate_gas_cost(self, txn: TransactionAPI, **kwargs) -> int:
 
     @property
     def chain_id(self) -> int:
-        if self.network.name not in (
+        if hasattr(self.web3, "eth"):
```

Comment:
`self.web3` may raise `ProviderNotConnectedError`

Reply:
I think this is fine as long as `__repr__` does not error when disconnected, I think `__repr__` handles that though and we are tests.

I believe this works.
I am not sure what is with the tests though.